sources/PTYTextView.m

```diff
@@ -1434,7 +1435,8 @@ - (void)keyDown:(NSEvent*)event {
         // track the instance of PTYTextView that is currently handling a key event and rerouting
         // calls as needed in -insertText and -doCommandBySelector.
         gCurrentKeyEventTextView = [[self retain] autorelease];
-        [self interpretKeyEvents:[NSArray arrayWithObject:event]];
+        _keydownEvent = event;
+        _inputMethodIsInserting = !event.isARepeat && [self.inputContext handleEvent:event];
```

Comment:
I'm using OS 10.11.4. The problem is still reproduced on this branch: https://github.com/tai2/iTerm2/commit/b8cf2bc0284fab5bef06d481e90ec0e68a1f6313 .